tests/test_sensor_camera.py

```diff
@@ -0,0 +1,168 @@
+        return tensor_or_array
+
+    for cam_name, data in [("cam0", data_cam0), ("cam1", data_cam1), ("attached", data_attached)]:
+        rgb = data.rgb
+        rgb_np = to_numpy_for_test(rgb)
+
+        # Check not pure black
+        mean_value = np.mean(rgb_np)
+        assert mean_value > 1.0, f"{cam_name}: Image is too dark (mean={mean_value:.2f}), likely pure black"
+
+        # Check not pure white
+        assert mean_value < 254.0, f"{cam_name}: Image is too bright (mean={mean_value:.2f}), likely pure white"
+
+        # Check variance (should have some texture/variation)
+        variance = np.var(rgb_np)
+        assert variance > 1.0, f"{cam_name}: Image has no variation (var={variance:.2f}), likely uniform color"
+
+    # Test read with envs_idx
+    if n_envs > 0:
+        data_env0 = raster_cam0.read(envs_idx=0)
+        assert data_env0.rgb.shape == (512, 512, 3), f"Single env read shape mismatch: got {data_env0.rgb.shape}"
+
+        data_env1 = raster_cam0.read(envs_idx=1)
+        assert data_env1.rgb.shape == (512, 512, 3), f"Single env read shape mismatch: got {data_env1.rgb.shape}"
+
+    # Helper to query the world position of a rasterizer camera sensor
+    def _get_camera_world_pos(sensor):
+        renderer = sensor._shared_metadata.renderer
+        context = sensor._shared_metadata.context
+        node = renderer._camera_nodes[sensor._idx]
+        pose = context._scene.get_pose(node)
+        # Handle possible batched pose (n_envs, 4, 4)
+        if pose.ndim == 3:
+            pose = pose[0]
+        return pose[:3, 3].copy()
+
+    # Store attached camera position before detachment
+    cam_pos_initial = _get_camera_world_pos(raster_cam_attached)
+
+    # Continue simulation to let sphere fall
+    for i in range(20):
+        scene.step()
+
+    # Read attached camera (should follow sphere; auto-render)
+    data_attached_moved = raster_cam_attached.read()
+
+    # Camera position should be different after sphere moved
+    cam_pos_moved = _get_camera_world_pos(raster_cam_attached)
+    cam_move_dist = np.linalg.norm(cam_pos_moved - cam_pos_initial)
+    assert cam_move_dist > 1e-3, f"Attached camera position didn't change after sphere moved (dist={cam_move_dist:.3e})"
+
+    # Store camera position at the moment of detachment
+    cam_pos_at_detach = cam_pos_moved.copy()
+
+    # Detach camera
+    raster_cam_attached.detach()
+
+    # Continue simulation
+    for i in range(20):
+        scene.step()
+
+    # After detachment, camera should stay at same position while sphere continues falling
+    # So the camera position should be (almost) unchanged
+    cam_pos_after_detach = _get_camera_world_pos(raster_cam_attached)
+    cam_move_after_detach = np.linalg.norm(cam_pos_after_detach - cam_pos_at_detach)
+    assert (
+        cam_move_after_detach < 1e-6
+    ), f"Detached camera position changed too much (dist={cam_move_after_detach:.3e}), should stay static"
```

Comment:
This entire section should be deleted.

Reply:
> you mean we shouldn't provide detach()? why?

This is not part of the sensor API. All sensors should have the exact same API. No exception.